core/src/main/java/org/jdbi/v3/core/mapper/MapMapper.java

```diff
@@ -16,19 +16,29 @@
 import java.sql.ResultSet;
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
+import java.util.ArrayList;
 import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Locale;
 import java.util.Map;
-
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import org.jdbi.v3.core.config.JdbiConfig;
 import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.meta.Beta;
 
 /**
  * Yo dawg, I heard you like maps, so I made you a mapper that maps rows into {@code Map<String,Object>}. Map
  * keys are column names, while map values are the values in those columns. Map keys are converted to lowercase by
  * default.
  */
 public class MapMapper implements RowMapper<Map<String, Object>> {
-    private final boolean foldCase;
+    /**
+     * @deprecated TODO remove in jdbi4
+     */
+    @Deprecated
+    private final boolean toLowerCase;
```

Comment:
Simpler to make this a `Boolean`, and have this field take precedence if non-null.

Reply:
done